.buildkite/pipeline.py

```diff
@@ -0,0 +1,431 @@
+            "instanceType": self.instance_type,
+        }
+
+
+class GCPAgent(Agent):
+    """GCP Agent object"""
+
+    def __init__(self, image: str):
+        super().__init__(image, "gcp")
+
+    def create_entity(self) -> dict[str, str]:
+        return {
+            "provider": self.provider,
+            "image": self.image,
+        }
+
+
+class OrkaAgent(Agent):
+    """Orka Agent object"""
+
+    def __init__(self, image: str):
+        super().__init__(image, "orka")
+
+    def create_entity(self) -> dict[str, str]:
+        return {
+            "provider": self.provider,
+            "imagePrefix": self.image,
+        }
+
+
+class Step:
+    """Buildkite Step object"""
+
+    def __init__(
+        self,
+        name: str,
+        project: str,
+        category: str,
+        agent: Agent,
+        definition: dict[str, Any],
+    ):
+        self.command = definition.get("command", "")
+        self.env = definition.get("env", {})
+        self.agent: Agent = agent
+        self.name: str = name
+        self.project: str = project
+        self.category: str = category
+        self.comment = "/test " + self.project + " " + self.name
+        self.label = self.name
+
+    def __lt__(self, other):
+        return self.name < other.name
+
+    def step_command(self) -> list[str]:
+        commands = [
+            f"cd {self.project}",
+            self.command,
+        ]
+        return commands
+
+    def create_entity(self) -> dict[str, Any]:
+        data = {
+            "label": f"{self.project} {self.name}",
+            "command": self.step_command(),
+            "notify": [
+                {
+                    "github_commit_status": {
+                        "context": f"{self.project.title()}: {self.name}",
+                    }
+                }
+            ],
+            "agents": self.agent.create_entity(),
+            "artifact_paths": [
+                f"{self.project}/build/*.xml",
+                f"{self.project}/build/*.json",
+            ],
+        }
+        if self.env:
+            data["env"] = self.env
+        return data
+
+
+class Group:
+    """Buildkite Group object"""
+
+    def __init__(self, project: str, category: str, steps: list[Step]):
+        self.project: str = project
+        self.category: str = category
+        self.steps: list[Step] = steps
+
+    def __lt__(self, other):
+        return self.project < other.project
+
+    def create_entity(self) -> dict[str, Any]:
+        if len(self.steps) == 0:
+            return {}
+
+        data = {
+            "group": f"{self.project} {self.category}",
+            "key": f"{self.project}-{self.category}",
+            "steps": [step.create_entity() for step in self.steps],
+        }
+
+        return data
+
+
+class GitHelper:
+    def __init__(self):
+        self.files: list[str] = []
+
+    def get_pr_changeset(self) -> list[str]:
+        base_branch = os.getenv("BUILDKITE_PULL_REQUEST_BASE_BRANCH", "main")
```

Comment:
Will this hardcoded value cause any issues when running on building for non-PRs? Such as `7.17`, `8.13` ,... ?

Reply:
In theory no, because this function should only be called in the context of a PR.